pkg/virt-controller/watch/vm.go

```diff
@@ -734,6 +736,40 @@ func (c *VMController) startStop(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualM
 	}
 }
 
+// isVMIStartExpected determines whether a VMI is expected to be started for this VM.
+func (c *VMController) isVMIStartExpected(vm *virtv1.VirtualMachine) bool {
+	vmKey, err := controller.KeyFunc(vm)
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error fetching vmKey: %v", err)
+		return false
+	}
+
+	expectations, exists, _ := c.expectations.GetExpectations(vmKey)
+	if !exists || expectations == nil {
+		return false
+	}
+
+	adds, _ := expectations.GetExpectations()
+	return adds > 0
+}
+
+// isVMIStopExpected determines whether a VMI is expected to be stopped for this VM.
+func (c *VMController) isVMIStopExpected(vm *virtv1.VirtualMachine) bool {
+	vmKey, err := controller.KeyFunc(vm)
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error fetching vmKey: %v", err)
+		return false
+	}
+
+	expectations, exists, _ := c.expectations.GetExpectations(vmKey)
+	if !exists || expectations == nil {
+		return false
+	}
+
+	_, dels := expectations.GetExpectations()
+	return dels > 0
+}
+
```

Comment:
This is quite the optimization.

Reply:
In the sense that it's not needed? It basically allows to detect a status change a bit earlier.

Reply:
yup, i think it's fine. i was merely surprised to see it. Using expectaions like this was something I hadn't considered before